web/job_specs_controller_test.go

```diff
@@ -91,7 +91,17 @@ func TestJobSpecsController_Create(t *testing.T) {
 	app, cleanup := cltest.NewApplication()
 	defer cleanup()
 
-	j := cltest.FixtureCreateJobViaWeb(t, app, "../internal/fixtures/web/hello_world_job.json")
+	resp := cltest.BasicAuthPost(
+		app.Server.URL+"/v2/specs",
+		"application/json",
+		bytes.NewBuffer(cltest.LoadJSON("../internal/fixtures/web/hello_world_job.json")),
+	)
+	defer resp.Body.Close()
```

Comment:
The response body is closed for requests that return a status code `>= 300 & < 600` as part of [`ParseResponseBody`](https://github.com/smartcontractkit/chainlink/blob/master/internal/cltest/cltest.go#L337) which is called during `BasicAuthPost`.

How do you feel about closing the body for all requests instead?

Reply:
So it looks like we close the body in that case because we are erroring and expect the test not to proceed. If we close it for all requests in that method, it might be closed for other helpers that want to access it, which causes the test to panic.

I think we should probably just have one definitive place where we always close the body, instead of tucking it down in helpers. Maybe the helpers like BasicAuthPost and BasicAuthGet could return a cleanup callback that each test calls, and all other helpers assume the cleanup is taken care of?

Reply:
👍 I like the idea of the returned cleanup function